server.js

```diff
@@ -71,6 +73,165 @@ server.route([
+          SIGNATURE: process.env.PAYPAL_SIGNATURE,
+          METHOD: "GetExpressCheckoutDetails",
+          VERSION: "106.0",
+          TOKEN: request.url.query.token
+        }
+      }, function(err, httpResponse, body) {
+        if (err) {
+          return console.error('donation failed:', err);
+        }
+        var data = querystring.parse(body);
+        httpRequest({
+          url:'https://api-3t.sandbox.paypal.com/nvp',
+          method: "POST",
+          form: {
+            USER: process.env.PAYPAL_USER,
+            PWD: process.env.PAYPAL_PWD,
+            SIGNATURE: process.env.PAYPAL_SIGNATURE,
+            METHOD: "DoExpressCheckoutPayment",
+            VERSION: "106.0",
+            TOKEN: data.TOKEN,
+            PAYERID: data.PAYERID,
+            PAYMENTREQUEST_0_AMT: data.PAYMENTREQUEST_0_AMT,
+            PAYMENTREQUEST_0_CURRENCYCODE: data.CURRENCYCODE
+          }
+        }, function(err, httpResponse, body) {
+          if (err) {
+            return console.error('donation failed:', err);
+          }
+          var bodyData = querystring.parse(body);
+          reply.redirect('/thank-you/?tx=' + bodyData.PAYMENTINFO_0_TRANSACTIONID + '&amt=' + data.PAYMENTREQUEST_0_AMT + '&cc=' + data.CURRENCYCODE);
+        });
+      });
+    }
+  }, {
+    method: 'GET',
+    path: '/paypal-recurring-redirect',
+    handler: function(request, reply) {
+      httpRequest({
+        url:'https://api-3t.sandbox.paypal.com/nvp',
+        method: "POST",
+        form: {
+          USER: process.env.PAYPAL_USER,
+          PWD: process.env.PAYPAL_PWD,
+          SIGNATURE: process.env.PAYPAL_SIGNATURE,
+          METHOD: "GetExpressCheckoutDetails",
+          VERSION: "106.0",
+          TOKEN: request.url.query.token
+        }
+      }, function(err, httpResponse, body) {
+        if (err) {
+          return console.error('donation failed:', err);
+        }
+        var data = querystring.parse(body);
+        httpRequest({
+          url:'https://api-3t.sandbox.paypal.com/nvp',
+          method: "POST",
+          form: {
+            USER: process.env.PAYPAL_USER,
+            PWD: process.env.PAYPAL_PWD,
+            SIGNATURE: process.env.PAYPAL_SIGNATURE,
+            METHOD: "CreateRecurringPaymentsProfile",
+            VERSION: "106.0",
+            TOKEN: data.TOKEN,
+            PAYERID: data.PAYERID,
+            DESC: data.DESC,
+            PROFILESTARTDATE: data.TIMESTAMP,
+            BILLINGPERIOD: "Month",
+            BILLINGFREQUENCY: "12",
+            AMT: data.AMT,
+            INITAMT: data.AMT,
+            CURRENCYCODE: data.CURRENCYCODE
+          }
+        }, function(err, httpResponse, body) {
+          if (err) {
+            return console.error('donation failed:', err);
+          }
+          var bodyData = querystring.parse(body);
+          // The below TRANSACTIONID is undefined.
+          // I'm not sure how to get this yet for recurring payments.
```

Comment:
https://github.com/mozilla/donate.mozilla.org/issues/119

It's kinda low priority compared to other issues.